.github/actions/setup-test-cluster/action.yaml

```diff
@@ -0,0 +1,215 @@
+name: Set Up Test Cluster
+description: Set up Turing Test Cluster
+
+inputs:
+  turing_api_tar_archive_name:
+    required: true
+    description: 'Turing API tar Archive Name'
+    default: ''
+  turing_router_tar_archive_name:
+    required: true
+    description: 'Turing Router tar Archive Name'
+    default: ''
+  experiment_engine_plugin_archive_name:
+    required: true
+    description: 'Experiment Engine tar Archive Name'
+    default: ''
+  cluster_init_tar_archive_name:
+    required: true
+    description: 'Cluster init tar Archive Name'
+    default: ''
+  matrix_use_in_cluster_config:
+    required: true
+    description: 'Matrix: useInClusterConfig'
+    default: ''
+  matrix_values_file:
+    required: true
+    description: 'Matrix: valuesFile'
+    default: ''
+  cluster_name:
+    required: true
+    description: 'Name of Cluster'
+    default: ''
+  istio_version:
+    required: true
+    description: 'Istio Version'
+    default: ''
+  knative_version:
+    required: true
+    description: 'Knative Version'
+    default: ''
+  knative_istio_version:
+    required: true
+    description: 'Knative Istio Version'
+    default: ''
+  local_registry:
+    required: true
+    description: 'Endpoint of local registry'
+    default: ''
+  cluster_init_version:
+    required: true
+    description: 'Version of cluster to install, tar file has to follow naming - cluster-init.(CLUSTER_INIT_VERSION).tar '
+    default: ''
+
+runs:
+  using: composite
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ inputs.go_version }}
+
+    - name: Download Turing API Docker tar archive
+      uses: actions/download-artifact@v2
+      with:
+        name: ${{ inputs.turing_api_tar_archive_name }}
+
+    - name: Download Turing Router Docker tar archive
+      uses: actions/download-artifact@v2
+      with:
+        name: ${{ inputs.turing_router_tar_archive_name }}
+
+    - name: Download Experiment Engine Plugin Docker tar archive
+      uses: actions/download-artifact@v2
+      with:
+        name: ${{ inputs.experiment_engine_plugin_archive_name }}
+
+    - name: Download Cluster Init Docker tar archive
+      uses: actions/download-artifact@v2
+      with:
+        name: ${{ inputs.cluster_init_tar_archive_name }}
+
+    - name: Run action cluster-init
+      uses: ./.github/actions/run-cluster-init
+      with:
+        cluster_name: ${{ env.CLUSTER_NAME}}
```

Comment:
Since these are the inputs required by the Set Up Test Cluster action, it would be better to pass these values down from the calling workflow:

```yaml
# .github/workflows/turing.yaml
      - name: Set Up Test Cluster
        uses: ./.github/actions/setup-test-cluster
        with:
          go-version: ${{ env.GO_VERSION }}
          ...
          matrix_values_file: ${{ matrix.valuesFile }}
          cluster_name: ${{ env.CLUSTER_NAME}}
          ...
```

Reply:
Hmm is there a general rule when deciding when we'll pass these values as inputs from the calling workflow (as shown in your example) or to access these values directly as environment variables? There are quite a number of other values that are accessed in the Set Up Test Cluster action (and other actions as well) as environment variables, e.g. `DOCKER_REPOSITORY`, `TURING_ROUTER_VERSION`, `CLUSTER_INIT_VERSION`, etc.